lib/getOptions.js

```diff
@@ -156,26 +161,28 @@ var getBaseOptions = function(cb){
     options.flag_name = process.env.COVERALLS_FLAG_NAME;
   }
 
-  if (git_commit){
-    fetchGitData({
-      head: {
-        id: git_commit,
-        committer_name: git_committer_name,
-        committer_email: git_committer_email,
-        message: git_message
-      },
-      branch: git_branch
-    }, function(err, git){
-      if (err){
-        logger.warn('there was an error getting git data: ', err);
-      } else {
-        options.git = git;
-      }
-      return cb(err, options);
-    });
-  } else {
+  if (!git_commit){
     return cb(null, options);
   }
+  return fetchGitData({
+    head: {
+      id: git_commit,
+      committer_name: git_committer_name,
+      committer_email: git_committer_email,
+      message: git_message
+    },
+    branch: git_branch
+  }, function addGitToOptions(err, git){
+    if (err){
+      logger.warn('there was an error getting git data: ', err);
+    } else {
+      options.git = git;
+      if (options.parallel && !options.service_number) {
+        options.service_number = git_commit;
```

Comment:
I'm concerned about using the git commit sha here, in instances where CI tests are rerun on a failing test due to flaky tests could result in weird coverage data on the same coveralls build.

Reply:
I'm concerned about that, too.  However, this setting is a last-resort: the hope is that `npm-coveralls` knows a setting that's more appropriate to the service being used.  Personally, I only know Circle and Travis, so that's what I configured.